components/SpotCard.js

```diff
@@ -1,15 +1,57 @@
 import styled from "styled-components";
 import Image from "next/image";
 import placeholderSpotImage from "../assets/images/card-image-placeholder.jpg";
+import { Button } from "./Button";
+import SpotDetails from "./SpotDetails";
+import { useState } from "react";
+
+export default function SpotCard({ spot }) {
+  const [state, setState] = useState("hide");
 
-export default function SpotCard({spot}) {
   return (
     <Card>
       <SpotImage src={placeholderSpotImage} alt="Placeholder Spot Image" />
       <SpotInfos>
         <SpotCategory>{spot.category}</SpotCategory>
         <SpotName>"{spot.name}"</SpotName>
+        <Button
+          type="button"
+          variant="details"
+          onClick={() => {
+            setState(() => {
+              if (state === "hide") {
+                return "show";
+              } else {
+                return "hide";
+              }
+            });
```

Comment:
```suggestion
            setState((prevState) => {
              if (prevState === "hide") {
                return "show";
              } else {
                return "hide";
              }
            });
```
Since you are using the updater function within the setState function, it would be more consistent to use the given state (which I named `prevState` here) instead of relying on `state`.